python/common/org/python/types/List.java

```diff
@@ -167,8 +168,14 @@ public List(org.python.Object [] args, java.util.Map<java.lang.String, org.pytho
             int count = Math.min(size, otherSize);
 
             for (int i = 0; i < count; i++) {
-                org.python.types.Bool b =
-                    (org.python.types.Bool) this.value.get(i).__lt__(otherList.value.get(i));
+                org.python.Object result = this.value.get(i).__lt__(otherList.value.get(i));
```

Comment:
With respect to 2. (I know you CC'd @freakboy3742, but I think I have a possible answer, if he disagrees he can correct me).

In Python source code, `PyObject_RichCompare(Bool)` is in `object.c`. To me, this signifies that the method should probably be a static method `RichCompare(Bool)` on `org.python.types.Object`.

Also, from the current list of `.java` files, I can't find a better location.